src/js/angular/controllers.js

```diff
@@ -931,6 +840,50 @@ function mainCtrl($scope, $menuItems, $jwtAuth, $http, toastr, $location, $repos
 
         return $filter('date')(time, ("'" + $translate.instant('timestamp.on') + "' yyyy-MM-dd '" + $translate.instant('timestamp.at') + "' HH:mm"));
     };
+
+    // =========================
+    // Private functions
+    // =========================
+    const fibonacciGenerator = SequenceGeneratorUtil.getFibonacciSequenceGenerator();
+    const updateActiveOperations = () => {
```

Comment:
I wonder, shouldn't these be in the directive itself and not here

Reply:
When I implemented it, I thought about the directive as a component for visualizing a list of operations. The main controller is the client of this component, responsible for fetching and managing the data.

Reply:
Refactored